lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -0,0 +1,105 @@
+from flask_restful import Resource, abort
+from flask import jsonify
+
+from pbench.server.api.resources.query_apis import get_index_prefix
+
+
+class EndpointConfig(Resource):
+    """
+    EndpointConfig API resource: this supports dynamic dashboard configuration
+    from the Pbench server rather than constructing a disconnected dashboard
+    config file.
+    """
+
+    def __init__(self, config, logger):
+        """
+        __init__ Construct the API resource
+
+        Args:
+            config (PbenchServerConfig): server config values
+            logger (Logger): message logging
+        """
+        self.logger = logger
+        self.host = config.get("pbench-server", "proxy_host")
+        uri_prefix = config.rest_uri
+        self.uri = f"{self.host}{uri_prefix}"
```

Comment:
Might it be worthwhile to have a little code which ensures that there is exactly the right amount of punctuation between `self.host` and `uri_prefix`? (I.e., there's supposed to be exactly one `/` there -- right? -- provided by the prefix and _not_ the host?...but, what if it's missing from one or there's and extra from the other?)

Reply:
I'm just concerned about the case where "them" is "us" (or "me"!).  :-D  (It seems like we lost a bunch of time in our most recent Jam Session due to having the wrong number of slashes...as I recall, we were installing a reverse proxy and, apparently, we didn't quite know what we were doing....)

I understand and concur with resisting feature creep.  I s'pose I'll open an issue.

Reply:
That typo was on the NGINX side, so better handling of errors in our config file wouldn't have helped... I'm also not sure if it was inherently wrong to have the trailing slash or if it just conflicted with something else in Pete's NGINX config.

Still, again, the point is valid, and giving help even for one-time configuration is always a good thing. But I would definitely prefer that we attack that strategically with some deliberation.

Reply:
Perhaps we could use Python 3's `urllib` module?  See https://stackoverflow.com/questions/15799696/how-to-build-urls-in-python

Reply:
We can use `X-Forwarded-For` to find the original target URL, and then add our endpoints relative to that.  See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host.

Reply:
I've used `urllib` before ... I somehow didn't even think of it here because the construction is simple, but I like that suggestion.

As for the `X-Forwarded-For` ... I had wondered if there was a header like that but kept forgetting to investigate. Yeah, that's great, and I'll have to experiment with it. I like the idea of just configuring all the endpoints based on how the browser got to the configuration API. No header means local proxy, otherwise we use the header value.

Reply:
The urllib ... "limitations" didn't make it quite as useful as I'd hoped. It's really not well suited to gluing together sub-paths.

The new commit switches completely to dynamically detecting a reverse proxy by using either `X-Forwarded-Host` or `Forwarded` headers, and there are unit tests to check the handling.

There are some debug log statements here I'll remove, and I need to think about how to deal with the templated user URI, so I'm not quite ready for another round of reviews ... but I wanted to push a checkpoint to show where it it.